src/core/border_router/infra_if.hpp

```diff
@@ -201,6 +216,7 @@ class InfraIf : public InstanceLocator
     InfoString ToString(void) const;
 
 private:
+    otInstance *GetInstancePtr(void) const { return reinterpret_cast<otInstance *>(&InstanceLocator::GetInstance()); }
```

Comment:
This may not work under all configs and requires a forced `reinterpret_cast()` which is not desirable).

If we move the `GetLinkLayerAddress()` in `cpp` file we wont need this.